roles/kubernetes-apps/external_cloud_controller/openstack/tasks/openstack-credential-check.yml

```diff
@@ -4,24 +4,63 @@
     msg: "external_openstack_auth_url is missing"
   when: external_openstack_auth_url is not defined or not external_openstack_auth_url
 
-- name: External OpenStack Cloud Controller | check external_openstack_username value
+
+- name: External OpenStack Cloud Controller | check external_openstack_username or external_openstack_application_credential_name value
+  fail:
+    msg: "you must either set external_openstack_username or external_openstack_application_credential_name"
+  when:
+    - external_openstack_username is not defined or not external_openstack_username
+    - external_openstack_application_credential_name is not defined or not external_openstack_application_credential_name
+
+
+- name: External OpenStack Cloud Controller | check external_openstack_application_credential_id value
+  fail:
+    msg: "external_openstack_application_credential_id is missing"
+  when:
+    - external_openstack_application_credential_name is defined
+    - external_openstack_application_credential_name|length > 0
+    - external_openstack_application_credential_id is not defined or not external_openstack_application_credential_id
```

Comment:
This logic does not make sense to me. It functions like this:
If external_openstack_application_credential_name is defined, and it has a length > 0, and it is not defined or it is boolean False, then fail.
So, this will only fail if the var is somehow defined as boolean False, and if 'False' is considered to have a length >0. 

Are you aware that a list of when conditions is a logical AND, not OR ?

Reply:
Sorry I misread this part, nevermind.